nmdc_automation/re_iding/base.py

```diff
@@ -728,12 +762,79 @@ def _update_omics_processing_alternative_identifiers(omics_processing: nmdc.Omic
+        new_data_object_id = get_new_nmdc_id(updated_data_object, api_client, identifiers_map)
+        updated_data_object.id = new_data_object_id
+
+    return updated_data_object
+
+
+def update_metabolomics_analysis_activity(metabolomics_analysis_activity: nmdc.MetabolomicsAnalysisActivity,
+                                        nmdc_omics_processing_id: str, nmdc_input_data_object_id: str,
+                                          nmdc_output_data_object_ids: List[str], nmdc_calibration_data_object_id: str,
+                                         api_client: NmdcRuntimeApi, identifiers_map: dict=None) -> nmdc.MetabolomicsAnalysisActivity:
+    """
+    Update the metabolomics analysis activity record with the new ID
+    """
+    updated_metabolomics_analysis_activity = deepcopy(metabolomics_analysis_activity)
+    # Ensure that type is set to nmdc:MetabolomicsAnalysisActivity
+    updated_metabolomics_analysis_activity.type = "nmdc:MetabolomicsAnalysisActivity"
+
+    # Check if we need to update the metabolomics analysis activity ID
+    if not updated_metabolomics_analysis_activity.id.startswith("nmdc:wfmb-"):
+        new_metabolomics_analysis_activity_id = get_new_nmdc_id(updated_metabolomics_analysis_activity, api_client, identifiers_map)
+        updated_metabolomics_analysis_activity.id = new_metabolomics_analysis_activity_id
+
+    updated_metabolomics_analysis_activity.was_informed_by = nmdc_omics_processing_id
+    updated_metabolomics_analysis_activity.has_input = [nmdc_input_data_object_id]
+    updated_metabolomics_analysis_activity.has_output = nmdc_output_data_object_ids
+    updated_metabolomics_analysis_activity.has_calibration = nmdc_calibration_data_object_id
+
+    return updated_metabolomics_analysis_activity
+
+
+def update_nom_analysis_activity(nom_analysis_activity: nmdc.NomAnalysisActivity,
+                                 nmdc_omics_processing_id: str, nmdc_input_data_object_id: str,
+                                 nmdc_output_data_object_ids: List[str], api_client: NmdcRuntimeApi,
+                                 identifiers_map: dict=None) -> nmdc.NomAnalysisActivity:
+    """
+    Update the NOM analysis activity record with the new ID
+    """
+    updated_nom_analysis_activity = deepcopy(nom_analysis_activity)
+    # Ensure that type is set to nmdc:NomAnalysisActivity
+    updated_nom_analysis_activity.type = "nmdc:NomAnalysisActivity"
+
+    # Check if we need to update the NOM analysis activity ID
+    if not updated_nom_analysis_activity.id.startswith("nmdc:wfna-"):
+        new_nom_analysis_activity_id = get_new_nmdc_id(updated_nom_analysis_activity, api_client, identifiers_map)
+        updated_nom_analysis_activity.id = new_nom_analysis_activity_id
+
+    updated_nom_analysis_activity.was_informed_by = nmdc_omics_processing_id
+    updated_nom_analysis_activity.has_input = [nmdc_input_data_object_id]
+    updated_nom_analysis_activity.has_output = nmdc_output_data_object_ids
```

Comment:
has_output should be an array, not a string.

Reply:
The input argument `nmdc_output_data_object_ids` is a list.
```python
def update_nom_analysis_activity(nom_analysis_activity: nmdc.NomAnalysisActivity,
                                 nmdc_omics_processing_id: str, nmdc_input_data_object_id: str,
                                 nmdc_output_data_object_ids: List[str], api_client: NmdcRuntimeApi,
                                 identifiers_map: dict=None) -> nmdc.NomAnalysisActivity:
```